src/sentry/api/endpoints/team_time_to_resolution.py

```diff
@@ -0,0 +1,57 @@
+from collections import defaultdict
+from datetime import timedelta
+
+from django.db.models import Avg, F
+from django.db.models.functions import TruncDay
+from rest_framework.response import Response
+
+from sentry.api.base import EnvironmentMixin
+from sentry.api.bases.team import TeamEndpoint
+from sentry.api.utils import get_date_range_from_params
+from sentry.models import GroupHistory, GroupHistoryStatus, Project
+
+
+class TeamTimeToResolutionEndpoint(TeamEndpoint, EnvironmentMixin):
+    def get(self, request, team):
+        """
+        Return a a time bucketed list of mean group resolution times for a given team.
+        """
+        project_list = Project.objects.get_for_team_ids(team_ids=[team.id])
+        start, end = get_date_range_from_params(request.GET)
+        history_list = (
+            GroupHistory.objects.filter(
+                status=GroupHistoryStatus.RESOLVED,
+                project__in=project_list,
+                date_added__gte=start,
+                date_added__lte=end,
+            )
+            .annotate(bucket=TruncDay("date_added"))
+            .values("bucket", "prev_history_date")
+            .annotate(ttr=F("date_added") - F("prev_history_date"))
+            .annotate(avg_ttr=Avg("ttr"))
+        )
+        sums = defaultdict(dict)
+        for gh in history_list:
+            key = str(gh["bucket"])
+            if "sum" in sums[key]:
+                sums[key]["sum"] += gh["ttr"]
+                sums[key]["count"] += 1
+            else:
+                sums[key] = {"sum": gh["ttr"], "count": 1}
+        current_day = start.replace(hour=0, minute=0, second=0, microsecond=0) + timedelta(days=1)
```

Comment:
I don't quite follow why we add a day here

Reply:
I did this because I was getting an extra result I wanted to chop off (querying for 14d gave me 15 results, no time range queries were returning 91 instead of 90, etc)

Reply:
I think this means we're querying a little more than we should. If today is 2021-09-15 13:00, and you request 14d worth of stats, then it will use 2021-09-01 13:00 - 2021-09-15 13:00. But I guess what you want is 2021-09-02 - 2021-09-15 inclusive?

Wondering if we should round the start date up to the nearest day, and the end date down?

Reply:
Aaah yeah the idea of rounding end date down and start date up and _before_ we query is good...and I'm going to retroactively apply that to the other PRs that use the same logic as well.